packages/core/src/sync-store/postgres/store.ts

```diff
@@ -270,32 +270,28 @@ export class PostgresSyncStore implements SyncStore {
       .where("address", "=", address)
       .where("topic0", "=", eventSelector)
       .where("blockNumber", "<=", upToBlockNumber)
+      .orderBy("blockNumber", "asc")
       .limit(pageSize);
 
     let cursor: bigint | undefined = undefined;
 
     while (true) {
       let query = baseQuery;
-
-      if (cursor) {
-        query = query.where("blockNumber", ">", cursor);
-      }
+      if (cursor !== undefined) query = query.where("blockNumber", ">", cursor);
 
       const batch = await this.db.wrap(
         { method: "getFactoryChildAddresses" },
         () => query.execute(),
       );
 
-      const lastRow = batch[batch.length - 1];
-      if (lastRow) {
-        cursor = lastRow.blockNumber;
-      }
-
       if (batch.length > 0) {
         yield batch.map((a) => a.childAddress);
       }
 
+      // If the batch is less than the page size, there are no more pages.
       if (batch.length < pageSize) break;
+      // Otherwise, set the cursor to the last block number in the batch.
+      cursor = batch[batch.length - 1].blockNumber;
```

Comment:
I'm probably just confused about the context but for my own understanding why does paginating like this work in this case? Like if u set the cursor to the last block num in the batch, how do you know that the next iteration is landing on the next row (like if theres multiple rows with the same blockNumber, the next time you run the query the first row you hit probably won't be the next row in the "page")

Reply:
Great call out, that was also a bug. Per DMs, should be fixed now by using `id` as the pagination cursor.